Source/WebCore/platform/graphics/cocoa/MediaPlaybackTargetContext.h

```diff
@@ -57,21 +56,42 @@ class MediaPlaybackTargetContext {
     MediaPlaybackTargetContext() = default;
     WEBCORE_EXPORT explicit MediaPlaybackTargetContext(RetainPtr<AVOutputContext>&&);
 
-    MediaPlaybackTargetContext(RetainPtr<NSData>&& serializedOutputContext, bool hasActiveRoute)
-        : m_type(Type::SerializedAVOutputContext)
-        , m_serializedOutputContext(WTFMove(serializedOutputContext))
-        , m_cachedHasActiveRoute(hasActiveRoute)
-    {
-        ASSERT(m_serializedOutputContext);
-    }
-
     MediaPlaybackTargetContext(const String& mockDeviceName, MockState state)
         : m_type(Type::Mock)
         , m_mockDeviceName(mockDeviceName)
         , m_mockState(state)
     {
     }
 
+    struct NonPlatformData {
+        String mockDeviceName;
+        MockState mockState;
+    };
+
+    struct PlatformData {
+        RetainPtr<AVOutputContext> outputContext;
+    };
+
+    using PlatformDataVariant = std::variant<struct PlatformData, struct NonPlatformData>;
+
+    MediaPlaybackTargetContext(bool requirePlatformData, Type type, std::optional<PlatformDataVariant>&& platformDataVariant)
+        : m_requirePlatformData(requirePlatformData)
+        , m_type(type)
+        {
+            if (platformDataVariant) {
+                WTF::switchOn(*platformDataVariant,
+                    [&] (const PlatformData& d) {
+                        MediaPlaybackTargetContext(d.outputContext.get());
```

Comment:
I believe you're constructing a new `MediaPlaybackTargetContext ` object here and destroying it right away.